tracer/src/Datadog.Trace/Iast/IastModule.cs

```diff
@@ -631,6 +632,11 @@ private static IastModuleResponse GetScope(string evidenceValue, IntegrationId i
         {
             if (isRequest)
             {
+                if (traceContext?.RootSpan is not null)
+                {
+                    Security.Instance?.SetTraceSamplingPriority(traceContext.RootSpan);
```

Comment:
nit: You can avoid the double access with pattern matching 

```suggestion
                if (traceContext?.RootSpan is { } rootSpan)
                {
                    Security.Instance?.SetTraceSamplingPriority(rootSpan);
```